src/components/cylc/tree/TreeItem.vue

```diff
@@ -38,200 +41,153 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
           {{ $options.icons.mdiChevronRight }}
         </v-icon>
       </v-btn>
-      <!-- the node value -->
-      <!-- TODO: revisit these values that can be replaced by constants later (and in other components too). -->
-      <slot name="cyclepoint" v-if="node.type === 'cycle'">
-        <div :class="nodeDataClass" @click="nodeClicked">
-          <!-- NOTE: cycle point nodes don't have any data associated with them
-            at present so we must use the root family node for the task icon.
-            We don't use this for the v-cylc-object as that would set the node
-            type to family. -->
-          <Task
-            v-cylc-object="node"
-            v-if="node.familyTree?.length"
-            :key="node.id"
-            :task="node.familyTree[0].node"
-          />
-          <span class="mx-1">{{ node.name }}</span>
-        </div>
-      </slot>
-      <slot name="family-proxy" v-else-if="node.type === 'family'">
-        <div :class="nodeDataClass" @click="nodeClicked">
-          <Task
-            v-cylc-object="node"
-            :key="node.id"
-            :task="node.node"
-          />
-          <span class="mx-1">{{ node.name }}</span>
-        </div>
-      </slot>
-      <slot name="task-proxy" v-else-if="node.type === 'task'">
-        <div :class="nodeDataClass" @click="nodeClicked">
-          <!-- Task summary -->
-          <Task
-            v-cylc-object="node"
-            :key="node.id"
-            :task="node.node"
-            :startTime="(latestJob(node) || {}).startedTime"
+      <slot v-bind="{ isExpanded }">
+        <!-- the node value -->
+        <!-- TODO: revisit these node.type values that can be replaced by constants later (and in other components too). -->
+        <div
+          :class="nodeDataClass"
+          :style="nodeDataStyle"
+        >
+          <template v-if="node.type === 'cycle'">
+            <!-- NOTE: cycle point nodes don't have any data associated with them
+              at present so we must use the root family node for the task icon.
+              We don't use this for the v-cylc-object as that would set the node
+              type to family. -->
+            <Task
+              v-cylc-object="node"
+              v-if="node.familyTree?.length"
+              :key="node.id"
+              :task="node.familyTree[0].node"
```

Comment:
I thought about it but decided to hold off. Performance-wise, I don't think this is too much of a problem because there is not likely to be too many workflows in GScan compared to cycle points/families/tasks in the Tree view which could be hundreds. (Having said that, we did see that one case of run249!)

Reply:
So, it probably should go there, but there's no pressure to do so ATM?

Reply:
Want me to open an issue for it?

Reply:
Can do, I was mostly just trying to understand the motive for the changes.